scripts/cmake/rustlang.cmake

```diff
@@ -85,6 +85,31 @@ function(build_rust_archives)
         list(APPEND RUST_BUILD_CARGO_ENV MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET})
     endif()
 
+    if(ANDROID)
+        if(CMAKE_HOST_SYSTEM_NAME STREQUAL Linux)
+            set(ANDROID_HOST_TAG linux-x86_64)
+        elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL Darwin)
+            set(ANDROID_HOST_TAG darwin-x86_64)
+        elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)
+            set(ANDROID_HOST_TAG windows-x86_64)
+        else()
+            message("Unsupported OS?")
+        endif()  
+        # Toolchain.
+        set(ANDROID_TOOLCHAIN_ROOT
+            "${CMAKE_ANDROID_NDK}/toolchains/llvm/prebuilt/${ANDROID_HOST_TAG}")
+       
+        # Rust and Clang disagree about armv7, otherwise we can take the Rust build arch. 
+        if(${RUST_BUILD_ARCH} STREQUAL armv7-linux-androideabi )
```

Comment:
```suggestion
        if(${RUST_BUILD_ARCH} STREQUAL armv7-linux-androideabi)
```

Reply:
I'll commi this myself and attempt a merge. I think this might change the situation I am in local development right now.